src/engine/tt_hash_table.h

```diff
@@ -46,40 +47,51 @@ class TtHashTable {
     using ProbeResult = std::pair<int32_t, movegen::Move>;
     constexpr static std::optional<ProbeResult> probe(uint64_t key, int32_t alpha, int32_t beta, uint8_t depth, uint8_t ply)
     {
-        auto& entry = s_ttHashTable[key % s_ttHashSize];
+        auto& entryWrapper = s_ttHashTable[key % s_ttHashSize];
+        // Load all at once atomically
+        auto oldEntry = entryWrapper.load(std::memory_order_acquire);
 
-        if (entry.key != key) {
+        if (oldEntry.key != key) {
             return std::nullopt;
         }
 
         /* Entry is outdated (from a previous generation) */
-        if (entry.generation != s_currentGeneration) {
-            entry.key = 0; // Invalidate the entry
+        if (oldEntry.generation != s_currentGeneration) {
+            TtHashEntry newEntry = oldEntry;
+            newEntry.key = 0;
             if (s_hashCount > 0) {
+                // TODO make atomic
+                // Not sure how?
                 --s_hashCount;
```

Comment:
Nice-to-have value for GUI only. Currently modifying within reader. Could be racy but not worth the overhead for a ballpark value.

Might lead to overflow, let's see if it does.